families/supplychain/python/sawtooth_supplychain/rest_api/route_handlers.py

```diff
@@ -0,0 +1,795 @@
+            raise errors.CountInvalid()
+
+        if not (count >= 0 and p_min >= 0):
+            LOGGER.debug("Invalid paging parameter.")
+            raise errors.InvalidPagingQuery()
+
+        if p_max is not None:
+            if p_max < 0:
+                LOGGER.debug("Invalid paging parameter.")
+                raise errors.InvalidPagingQuery()
+
+        return count, p_min, p_max
+
+    async def list_agents(self, request):
+        """Fetches a paginated list of all Agents. Using the name filter
+        parameter narrows the list to any Agents who have a name which
+        contains a specified case-insensitive string.
+
+        Request:
+            query:
+                - name: a partial name to filter Agents by
+                - count: Number of items to return
+                - min: Id or index to start paging (inclusive)
+                - max: Id or index to end paging (inclusive)
+
+        Response:
+            data: JSON array of Agents
+        """
+
+        # Get query parameters
+        name = request.url.query.get('name', '')
+        # default count of 1000 to prevent overly long requests
+        count = request.url.query.get('count', 1000)
+        p_min = request.url.query.get('min', 0)
+        p_max = request.url.query.get('max', None)
+
+        count, p_min, p_max = self._check_paging_params(count, p_min, p_max)
+
+        # Create query
+        query_tuple = ()
+        name = "%" + name + "%"  # Add pattern matching to search string
+
+        if p_max is not None:
+            query = ("SELECT identifier, name FROM agent WHERE name like %s "
+                     "AND id >= %s AND id <= %s ORDER BY id limit %s")
+            query_tuple += (name, p_min, p_max, count)
+        else:
+            query = ("SELECT identifier, name FROM agent WHERE name like %s "
+                     "AND id >= %s ORDER BY id limit %s")
+            query_tuple += (name, p_min, count)
+
+        # Get a connection and cursor
+        try:
+            conn = self._db_connect()
+        except psycopg2.OperationalError:
+            LOGGER.debug("Could not connect to database.")
+            raise errors.DatabaseConnectionError()
+
+        cur = conn.cursor()
+
+        # Execute the query
+        try:
+            cur.execute(query, query_tuple)
+        except:
+            LOGGER.debug("Could not execute query: %s", query)
+            raise errors.UnknownDatabaseError()
+
+        # Fetch the rows
+        rows = cur.fetchall()
+
+        # Close DB connection
+        try:
+            self._db_close(conn, cur)
+        except:
+            LOGGER.debug("Could not close database connection.")
+            raise errors.UnknownDatabaseError()
+
+        fields = ('identifier', 'name')
+        data = self._make_dict(fields, rows)
+
+        # Return the data
+        return self._wrap_response(
+            request,
+            data=data,
+            metadata="")
```

Comment:
You need to be including the paging-related metadata here.  See the Sawtooth REST API (specifically it's function `_wrap_paginated_response`)